modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/base_resource_test_case.ftl

```diff
@@ -355,15 +364,38 @@ public abstract class Base${schemaName}ResourceTestCase {
 			<#elseif javaMethodSignature.methodName?contains("Permission")>
 				@Test
 				public void test${javaMethodSignature.methodName?cap_first}() throws Exception {
-					<#if freeMarkerTool.hasPostSchemaJavaMethodSignature(javaMethodSignatures, javaMethodSignature.pathJavaMethodParameters[0].parameterName, schemaName)>
-						@SuppressWarnings("PMD.UnusedLocalVariable")
-						${schemaName} post${schemaName} = test${freeMarkerTool.getPostSchemaJavaMethodSignature(javaMethodSignatures, "siteId", schemaName).methodName?cap_first}_add${schemaName}(random${schemaName}());
+					${schemaName} post${schemaName} = test${javaMethodSignature.methodName?cap_first}_add${schemaName}();
 
-						Page<Permission> page = ${schemaVarName}Resource.${javaMethodSignature.methodName}(testGroup.getGroupId(), RoleConstants.GUEST);
+					Page<Permission> page = ${schemaVarName}Resource.${javaMethodSignature.methodName}(
+						<#if stringUtil.equals("assetLibraryId", javaMethodSignature.javaMethodParameters[0].parameterName)>
+							testDepotEntry.getDepotEntryId()
+						<#elseif stringUtil.equals("siteId", javaMethodSignature.javaMethodParameters[0].parameterName)>
+							testGroup.getGroupId()
+						<#else>
+							post${schemaName}.getId()
+						</#if>
+					, RoleConstants.GUEST);
+
+					Assert.assertNotNull(page);
+				}
 
-						Assert.assertNotNull(page);
+				protected ${schemaName} test${javaMethodSignature.methodName?cap_first}_add${schemaName}() throws Exception {
+					<#if freeMarkerTool.hasPostSchemaJavaMethodSignature(javaMethodSignatures, javaMethodSignature.javaMethodParameters[0].parameterName, schemaName)>
+						<#assign postSchemaJavaMethodSignature = freeMarkerTool.getPostSchemaJavaMethodSignature(javaMethodSignatures, javaMethodSignature.javaMethodParameters[0].parameterName, schemaName) />
+						return test${postSchemaJavaMethodSignature.methodName?cap_first}_add${schemaName}(random${schemaName}()
+							<#if freeMarkerTool.hasRequestBodyMediaType(postSchemaJavaMethodSignature, "multipart/form-data")>
```

Comment:
Another observation. We should add a task to refactor the duplicated code...

Reply:
I left this as it is here for now. Not sure how much time it was desirable to spend on this at the moment. We can go for this whenever you feel it is a good moment